compiler/rustc_mir/src/transform/simplify.rs

```diff
@@ -306,6 +307,11 @@ pub fn remove_dead_blocks(body: &mut Body<'_>) {
         }
         used_blocks += 1;
     }
+
+    if tcx.sess.instrument_coverage() {
```

Comment:
If we can either leave a comment explaining why we need to do this here or on the `save_unreachable_coverage` function, that would be great!

Reply:
No problem. I'll add it and I'll also squash the commits now that you've had a chance to review.